### PR TITLE
[KOB-4374] - Update UI and Logix for plugin

### PR DESCRIPTION
**Pull request by tiennguyenhoan**

### CODE REVIEW

File: src/main/resources/kobiton/ListAvailableDevices.py [hunk 1]
```diff
@@ -2,110 +2,135 @@
...
 }
 
-devices = {}
+def getAvailableDevices():
+  try:
+    devicesFetchingParams = getDevicesFetchingParams(customFetchingParams, platformOptions, groupId)
+    rawDevicesList = getDevicesList(devicesFetchingParams)
 
+    filteredDevicesList = filterDevices(deviceGroups, rawDevicesList) 
+    print str(filteredDevicesList)
```
Comment (duylam) [inline src/main/resources/kobiton/ListAvailableDevices.py]: Is `print` used for debugging only ? Please remove if yes

File: src/main/resources/kobiton/Server.py [hunk 2]
```diff
@@ -4,23 +4,24 @@
...
+  "server": "Api Server"
 }, {
   "Authorization": 'Basic %s' % base64.b64encode('%s:%s' % (configuration.username, configuration.apiKey))
 })
 
-ping({
-  "url": re.sub(r'\/$|\\$','',configuration.remoteServer) + '/ping',
-  "server": "Remote Server"
+verifyConnection({
+  "url": re.sub(r'\/$|\\$','',configuration.executorServer) + '/ping',
```
Comment (duylam) [inline src/main/resources/kobiton/Server.py:23]: Please add implemenetation for this endpoint to server

File: src/main/resources/synthetic.xml [hunk 3]
```diff
@@ -6,9 +6,9 @@
...
     <property name="username" label="Kobiton Username" category="input" required="true" description="Your Kobiton Username" />
-    <property name="apiKey" label="Kobiton API Key" password="true" category="input" required="true" description="Your Kobiton API Key" />
-    <property name="url" label="Kobiton API URL" category="input" required="true" default="https://api.kobiton.com" description="Kobiton URL used to consume API"/>
-    <property name="remoteServer" label="Remote server domain" category="input" default="http://localhost:4545" required="true" description="Your server domain for receiving requests." />
+    <property name="apiKey" label="Kobiton Api Key" password="true" category="input" required="true" description="Your Kobiton API Key" />
+    <property name="apiServer" label="Kobiton Api Url" category="input" required="true" default="https://api.kobiton.com" description="Kobiton URL used to consume API"/>
+    <property name="executorServer" label="Executor Server Url" category="input" default="https://executor.kobiton.com:4545" required="true" description="Executor server domain for receiving requests." />
```
Comment (duylam) [inline src/main/resources/synthetic.xml]: Please use standard port (443) for this, don't use 4545

File: src/main/resources/synthetic.xml [hunk 4]
```diff
@@ -18,59 +18,62 @@
...
       </enum-values>
     </property>
     <property name="appUrl" label="App URL/App ID" category="input" required="false" description="Your app's public URL or Kobiton App ID. Required if you chose to test on apps." />
-    <property name="groupId" label="Group ID" category="input" required="true" description="Your group ID." />
-    <property name="gitUrl" label="Git Repository" category="input" required="true" description="URL to your Git repository that contains automation test scripts. Go to http://localhost:4545 to verify your repository's accessibility." />
-    <property name="ssh" label="Git Repository SSH Key" category="input" required="false" size="large" description="SSH to access your private Git repository." />
+    <property name="groupId" label="Group ID" category="input" required="false" description="Your group ID." />
+    <property name="gitUrl" label="Git Repository" category="input" required="true" description="URL to your Git repository that contains automation test scripts." />
+    <property name="ssh" label="Private repository SSH Key" category="input" required="false" size="large" description="SSH to access your private Git repository." />
```
Comment (duylam) [inline src/main/resources/synthetic.xml:68]: Add one more param called `gitBranch`, default value is `master`

**Review by duylam — COMMENTED** [review]
Please help address comments

File: src/main/resources/kobiton/Server.py (hunk 2 above)
Comment (tiennguyenhoan) [reply src/main/resources/kobiton/Server.py:23]: It already have

**Comment by duylam** [comment]
LGTM, thanks @NguyenHoanTien for this work

**Pull request merged**